### PR TITLE
KAFKA-7370: Enhance FileConfigProvider to read a dir

### PR DESCRIPTION
**Pull request by rayokota**
This is a backward compatible enhancement to augment FileConfigProvider
to read from a directory, where the file names are the keys and the
corresponding file contents are the values.  The former functionality of
reading a Properties file is retained.

This will allow for easier integration with secret management systems
where each secret is often written to an individual file, such as in
some Docker and Kubernetes setups.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by rayokota** [comment]
@ewencp , @rhauch can you review this small enhancement?  Thanks!

File: clients/src/main/java/org/apache/kafka/common/config/provider/FileConfigProvider.java [hunk 1]
```diff
@@ -93,9 +129,16 @@ public ConfigData get(String path, Set<String> keys) {
         }
     }
 
-    // visible for testing
-    protected Reader reader(String path) throws IOException {
-        return Files.newBufferedReader(Paths.get(path));
+    private String toString(Path path) {
```
Comment (viktorsomogyi) [inline clients/src/main/java/org/apache/kafka/common/config/provider/FileConfigProvider.java]: I think in general we shouldn't call this `toString`. People would confuse it for the actual `toString` method and would suppose the same or similar behavior. For instance `toString` methods aren't expected to throw exceptions.
So in my opinion we should call this something like `readAll` or `readAllBytes`.

**Review by viktorsomogyi — COMMENTED** [review]
Hi @rayokota, I've reviewed your changes. I think it overall looks good but left one comment for discussion.

**Comment by rayokota** [comment]
@viktorsomogyi , thanks for the review!  I've made the change as you suggested.

**Pull request closed without merging**